Go tests failing on Mac due to inconsistent Go error message

When running the test suite on mac I get the following error:

```
2022/07/18 15:15:15 http: TLS handshake error from 127.0.0.1:55752: remote error: tls: bad certificate
--- FAIL: TestFindChartInAuthAndTLSAndPassRepoURL (0.02s)
    chartrepo_test.go:314: Expected TLS error for function  FindChartInAuthAndTLSAndPassRepoURL not found, but got a different error (looks like "https://127.0.0.1:55750" is not a valid chart repository or cannot be reached: Get "https://127.0.0.1:55750/index.yaml": x509: “Acme Co” certificate is not trusted)
```

This is due to the mac platform provider having a different response from other platforms. In the golang source they even test for different messages from [mac](https://github.com/golang/go/blob/go1.18.4/src/crypto/x509/root_darwin_test.go#L56) and [windows](https://github.com/golang/go/blob/go1.18.4/src/crypto/x509/root_windows_test.go#L68).

<!-- If you need help or think you have found a bug, please help us with your issue by entering the following information (otherwise you can delete this text): -->

Output of `helm version`:
tip of main (developing for 3.10.0). Commit 937798868518b7c5dbcbefac5bfe0b2e045e3bae

Output of `kubectl version`: N/A

Cloud Provider/Platform (AKS, GKE, Minikube etc.):  N/A

--- a/pkg/repo/chartrepo_test.go
+++ b/pkg/repo/chartrepo_test.go
@@ -24,6 +24,7 @@ import (
 	"os"
 	"path/filepath"
 	"reflect"
+	"runtime"
 	"strings"
 	"testing"
 	"time"
@@ -309,8 +310,15 @@ func TestFindChartInAuthAndTLSAndPassRepoURL(t *testing.T) {
 
 	// If the insecureSkipTLsverify is false, it will return an error that contains "x509: certificate signed by unknown authority".
 	_, err = FindChartInAuthAndTLSAndPassRepoURL(srv.URL, "", "", "nginx", "0.1.0", "", "", "", false, false, getter.All(&cli.EnvSettings{}))
-
-	if !strings.Contains(err.Error(), "x509: certificate signed by unknown authority") {
+	// Go communicates with the platform and different platforms return different messages. Go itself tests darwin
+	// differently for its message. On newer versions of Darwin the message includes the "Acme Co" portion while older
+	// versions of Darwin do not. As there are people developing Helm using both old and new versions of Darwin we test
+	// for both messages.
+	if runtime.GOOS == "darwin" {
+		if !strings.Contains(err.Error(), "x509: “Acme Co” certificate is not trusted") && !strings.Contains(err.Error(), "x509: certificate signed by unknown authority") {
+			t.Errorf("Expected TLS error for function  FindChartInAuthAndTLSAndPassRepoURL not found, but got a different error (%v)", err)
+		}
+	} else if !strings.Contains(err.Error(), "x509: certificate signed by unknown authority") {
 		t.Errorf("Expected TLS error for function  FindChartInAuthAndTLSAndPassRepoURL not found, but got a different error (%v)", err)
 	}
 }

